### PR TITLE
Upgrade ibc-go to v7

### PR DESCRIPTION
**Pull request by dongrie**
* Upgrade ibc-go to v7
* Use latest yui-relayer (v0.4.0)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,121 +6,177 @@ require (
 	github.com/avast/retry-go v3.0.0+incompatible
 	github.com/btcsuite/btcd v0.22.1
 	github.com/btcsuite/btcutil v1.0.3-0.20201208143702-a53e38424cce
-	github.com/cosmos/cosmos-sdk v0.45.7
-	github.com/cosmos/ibc-go/v4 v4.0.0
+	github.com/cosmos/cosmos-sdk v0.47.0
+	github.com/cosmos/ibc-go/v7 v7.0.0
 	github.com/ethereum/go-ethereum v1.11.6
 	github.com/gogo/protobuf v1.3.3
```
Comment (siburu) [inline go.mod:12]: use `cosmos/gogoproto` instead of `gogo/protobuf`

File: go.mod [hunk 2]
```diff
@@ -6,121 +6,177 @@ require (
...
+	gopkg.in/ini.v1 v1.67.0 // indirect
 	gopkg.in/natefinch/npipe.v2 v2.0.0-20160621034901-c1b8fa8bdcce // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
+	nhooyr.io/websocket v1.8.6 // indirect
+	pgregory.net/rapid v0.5.5 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
```
Comment (siburu) [inline go.mod:182]: delete this replace directive.

File: go.mod (hunk 1 above)
Comment (dongrie) [reply go.mod]: use `cosmos/gogoproto`
https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/2/commits/21b222ebd3ee95cd7cdd146b6460e32badcbdeb9

File: go.mod (hunk 2 above)
Comment (dongrie) [reply go.mod]: delete replace
https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/2/commits/21b222ebd3ee95cd7cdd146b6460e32badcbdeb9

**Review by bluele — COMMENTED** [review]
Many thanks!
If you receive LGTM from @siburu, please squash the commits into meaningful ones before the merge.

**Comment by dongrie** [comment]
Thanks for the review!
squashed multiple commits

**Review by siburu — APPROVED** [review]

**Pull request merged**